aes.go

```diff
@@ -38,63 +30,226 @@ func NewAESCipher(key []byte) (cipher.Block, error) {
 	if err != nil {
 		return nil, err
 	}
-	return &aesCipher{c}, nil
+	var block cipher.Block
+	cbcSupported := loadCipher(kind, cipherModeCBC) != nil
+	ctrSupported := loadCipher(kind, cipherModeCTR) != nil
+	gcmSupported := loadCipher(kind, cipherModeGCM) != nil
+	aes := aesCipher{c}
+	switch {
+	case cbcSupported && ctrSupported && gcmSupported:
```

Comment:
I don't have a concrete idea, but wouldn't it be possible to somehow auto-generate this permutation? Otherwise, if we want to support more modes (e.g., XTS, which is already in x/crypto), the whole internal logic would need a change?

Reply:
@dagood also proposed something similar. Now is 2vs1, and the XTS argument is strong. I'll autogenerate those optional AES modes 👍.